File: python/29a_703A_MishkaAndGame.py
def main():
    m_cnt, c_cnt = 0, 0
    n = int(input())
    for _ in range(n):
        m, c = input().split()
        m, c = int(m), int(c)
        if m > c:
            m_cnt += 1
        elif c > m:
            c_cnt += 1
    if m_cnt > c_cnt:
        print("Mishka")
    elif c_cnt > m_cnt:
        print("Chris")
    else:
        print("Friendship is magic!^^")

if __name__ == "__main__":
    main()

File: cpp/29a_703A_MishkaAndGame.cpp
#include<bits/stdc++.h>

int main(){
    int n, m, c, m_cnt = 0, c_cnt = 0;
    std::cin>>n;
    for(int i = 0; i < n; i++){
        std::cin>>m>>c;
        if(m > c){
            m_cnt++;
        }
        else if(c > m){
            c_cnt++;
        }
    }
    if(m_cnt > c_cnt){
        std::cout<<"Mishka";
    }
    else if(m_cnt < c_cnt){
        std::cout<<"Chris";
    }
    else{
        std::cout<<"Friendship is magic!^^";
    }
    return 0;
}